### PR TITLE
gclib: update to new version and unblacklist

### PR DESCRIPTION
**Pull request by bernt-matthias**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Needed to get gffcompare running https://github.com/bioconda/bioconda-recipes/pull/12407 (I will remove the gclib changes from the other RP)

Test did not run locally for me since g++ was not found. But compiler is in build requirements.

### CODE REVIEW

File: recipes/gclib/chrono.patch [hunk 1]
```diff
@@ -0,0 +1,63 @@
+diff -ruN /tmp/gclib-0.10.3/GThreads.cpp ./GThreads.cpp
```
Comment (epruesse) [inline recipes/gclib/chrono.patch]: Can you put a comment at the top of this file, explaining briefly what it does? (Yep, that's allowed in patch files)

Comment (epruesse) [reply recipes/gclib/chrono.patch]: (And the other patches as well ... this one seems to need a lot of patching...)

**Comment by bernt-matthias** [comment]
I reported upstream: https://github.com/gpertea/gclib/issues/3

**Comment by bernt-matthias** [comment]
I added the comments. But @gpertea suggested to build gffcompare differently and NOT to distribute gclib as library: https://github.com/gpertea/gclib/issues/4

I will give it a try. Maybe we leave the PR open until this is done.

**Comment by bernt-matthias** [comment]
Superseded by https://github.com/bioconda/bioconda-recipes/pull/12407

**Pull request closed without merging**